src/NuGet.Core/NuGet.Protocol.Core.v3/Resources/PackageUpdateResource.cs

```diff
@@ -526,5 +560,53 @@ private bool IsV2LocalRepository(string root)
 
             return true;
         }
+
+        // Get a temp API key from nuget.org for pushing to https://nuget.smbsrc.net/
+        private async Task<string> GetSecureApiKey(
+            PackageIdentity packageIdentity,
+            string apiKey,
+            ILogger logger,
+            CancellationToken token)
+        {
+            if (string.IsNullOrEmpty(apiKey))
+            {
+                return apiKey;
+            }
+            var serviceEndpointUrl = GetServiceEndpointUrl(NuGetConstants.DefaultGalleryServerUrl,
+                string.Format(TempApiKeyServiceEndpoint, packageIdentity.Id, packageIdentity.Version));
+
+            try
+            {
+                var result = await _httpSource.GetJObjectAsync(
+                    new HttpSourceRequest(
+                        () =>
+                        {
+                            var request = HttpRequestMessageFactory.Create(
+                                HttpMethod.Put,
+                                serviceEndpointUrl,
+                                new HttpRequestMessageConfiguration(
+                                    logger: logger,
+                                    promptOn403: false));
+                            request.Headers.Add(ApiKeyHeader, apiKey);
+                            return request;
+                        }),
+                   logger,
+                   token);
+
+                return result.Value<string>("Key") ?? NullApiKey;
+            }
+            // If the package doesn't exist on nuget.org, nuget symbol server will ignore the api key.
+            // In this case, any api key should works, just set api key to some value here.
+            catch (HttpRequestException)
```

Comment:
What sort of HTTP status codes will end up here? It seems like anything but 200 and 403 should be rejected.